### ISSUE TITLE
SS-error

### ISSUE THREAD

**Issue by Frepke** [qdm12/gluetun#221]
**TLDR**: *shadowsocks: exec: "ss-server": executable file not found in $PATH*

1. Is this urgent?

    - [ ] Yes
    - [x] No

2. What VPN service provider are you using?

    - [ ] PIA
    - [ ] Mullvad
    - [ ] Windscribe
    - [x] Surfshark
    - [ ] Cyberghost

3. What's the version of the program?

    **See the line at the top of your logs**

    `Running version latest built on 2020-08-16T23:07:05Z (commit c59447c)`

4. What are you using to run the container?

    - [ ] Docker run
    - [x] Docker Compose
    - [ ] Kubernetes
    - [ ] Docker stack
    - [ ] Docker swarm
    - [ ] Podman
    - [ ] Other:

5. Extra information

Logs:

```
gluetun        | 🔧  Need help? https://github.com/qdm12/gluetun/issues/new
gluetun        | 💻  Email? quentin.mcgaw@gmail.com
gluetun        | ☕  Slack? Join from the Slack button on Github
gluetun        | 💸  Help me? https://github.com/sponsors/qdm12
gluetun        | 2020-08-17T10:41:02.158+0200   INFO    TinyProxy version: 1.10.0
gluetun        | 2020-08-17T10:41:02.163+0200   ERROR   exec: "ss-server": executable file not found in $PATH
gluetun        | 2020-08-17T10:41:02.183+0200   INFO    OpenVPN version: 2.4.9
gluetun        | 2020-08-17T10:41:02.192+0200   INFO    Unbound version: 1.10.1
gluetun        | 2020-08-17T10:41:02.195+0200   INFO    IPtables version: v1.8.4
gluetun        | 2020-08-17T10:41:02.196+0200   INFO    Settings summary below:
gluetun        | OpenVPN settings:
gluetun        | |--User: [redacted]
gluetun        | |--Password: [redacted]
gluetun        | |--Verbosity level: 1
gluetun        | |--Run as root: no
gluetun        | |--Surfshark settings:
gluetun        |  |--Network protocol: udp
gluetun        |  |--Region: netherlands amsterdam
gluetun        |  |--Target IP address: ������������Y.�h
gluetun        | |--Custom cipher: aes-256-gcm
gluetun        | System settings:
gluetun        | |--User ID: 1000
gluetun        | |--Group ID: 100
gluetun        | |--Timezone: europe/amsterdam
gluetun        | |--IP Status filepath: /ip
gluetun        | DNS over TLS disabled, using plaintext DNS 162.252.172.57
gluetun        | Firewall settings:
gluetun        |  |--Allowed subnets: 10.54.1.0/24
gluetun        |  |--VPN input ports: 
gluetun        | TinyProxy settings: disabled
gluetun        | ShadowSocks settings:
gluetun        |  |--Password: [redacted]
gluetun        |  |--Log: enabled
gluetun        |  |--Port: 8388
gluetun        |  |--Method: chacha20-ietf-poly1305
gluetun        | Public IP check period: 12h0m0s
gluetun        | 
gluetun        | 2020-08-17T10:41:02.212+0200   INFO    routing: default route found: interface eth0, gateway 172.17.0.1
gluetun        | 2020-08-17T10:41:02.212+0200   INFO    routing: local subnet found: 172.17.0.0/16
gluetun        | 2020-08-17T10:41:02.212+0200   INFO    openvpn configurator: checking for device /dev/net/tun
gluetun        | 2020-08-17T10:41:02.213+0200   WARN    TUN device is not available: open /dev/net/tun: no such file or directory
gluetun        | 2020-08-17T10:41:02.213+0200   INFO    openvpn configurator: creating /dev/net/tun
gluetun        | 2020-08-17T10:41:02.214+0200   INFO    Launching standard output merger
gluetun        | 2020-08-17T10:41:02.214+0200   INFO    firewall: enabling...
gluetun        | 2020-08-17T10:41:02.268+0200   INFO    firewall: enabled successfully
gluetun        | 2020-08-17T10:41:02.268+0200   INFO    firewall: setting allowed subnets through firewall...
gluetun        | 2020-08-17T10:41:02.276+0200   INFO    routing: adding 10.54.1.0/24 as route via 172.17.0.1 eth0
gluetun        | 2020-08-17T10:41:02.282+0200   INFO    shadowsocks configurator: generating configuration file
gluetun        | 2020-08-17T10:41:02.283+0200   INFO    http server: listening on 0.0.0.0:8000
gluetun        | 2020-08-17T10:41:02.284+0200   INFO    firewall: setting allowed input port 8388 through interface eth0...
gluetun        | 2020-08-17T10:41:02.283+0200   INFO    dns over tls: falling back on plaintext DNS at address 162.252.172.57
gluetun        | 2020-08-17T10:41:02.285+0200   INFO    dns configurator: using DNS address 162.252.172.57 internally
gluetun        | 2020-08-17T10:41:02.285+0200   INFO    dns configurator: using DNS address 162.252.172.57 system wide
gluetun        | 2020-08-17T10:41:02.296+0200   INFO    shadowsocks configurator: starting shadowsocks server
gluetun        | 2020-08-17T10:41:02.296+0200   ERROR   shadowsocks: exec: "ss-server": executable file not found in $PATH
gluetun        | 2020-08-17T10:41:02.296+0200   INFO    shadowsocks: retrying in 1 minute
gluetun        | 2020-08-17T10:41:02.296+0200   INFO    firewall: setting VPN connections through firewall...
gluetun        | 2020-08-17T10:41:02.304+0200   INFO    openvpn configurator: starting openvpn
gluetun        | 2020-08-17T10:41:02.310+0200   INFO    openvpn: OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020
gluetun        | 2020-08-17T10:41:02.310+0200   INFO    openvpn: library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
gluetun        | 2020-08-17T10:41:02.332+0200   INFO    openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]89.46.223.104:1194
gluetun        | 2020-08-17T10:41:02.332+0200   INFO    openvpn: UDP link local: (not bound)
gluetun        | 2020-08-17T10:41:02.332+0200   INFO    openvpn: UDP link remote: [AF_INET]89.46.223.104:1194
gluetun        | 2020-08-17T10:41:02.361+0200   INFO    openvpn: [nl-ams-v049.prod.surfshark.com] Peer Connection Initiated with [AF_INET]89.46.223.104:1194
gluetun        | 2020-08-17T10:41:03.477+0200   ERROR   openvpn: Unrecognized option or missing or extra parameter(s) in [PUSH-OPTIONS]:7: block-outside-dns (2.4.9)
gluetun        | 2020-08-17T10:41:03.479+0200   INFO    openvpn: TUN/TAP device tun0 opened
gluetun        | 2020-08-17T10:41:03.479+0200   INFO    openvpn: /sbin/ip link set dev tun0 up mtu 1500
gluetun        | 2020-08-17T10:41:03.482+0200   INFO    openvpn: /sbin/ip addr add dev tun0 10.8.8.2/24 broadcast 10.8.8.255
gluetun        | 2020-08-17T10:41:03.492+0200   INFO    openvpn: UID set to nonrootuser
gluetun        | 2020-08-17T10:41:03.492+0200   INFO    openvpn: Initialization Sequence Completed
gluetun        | 2020-08-17T10:41:03.492+0200   INFO    dns over tls: not restarting because disabled
gluetun        | 2020-08-17T10:41:03.493+0200   INFO    routing: default route found: interface eth0, gateway 172.17.0.1
gluetun        | 2020-08-17T10:41:03.494+0200   INFO    Gateway VPN IP address: 89.46.223.104
gluetun        | 2020-08-17T10:41:03.735+0200   INFO    ip getter: Public IP address is 89.46.223.105
gluetun        | 2020-08-17T10:42:02.296+0200   INFO    shadowsocks configurator: generating configuration file
gluetun        | 2020-08-17T10:42:02.297+0200   INFO    firewall: removing allowed port 8388 through firewall...
gluetun        | 2020-08-17T10:42:02.305+0200   INFO    firewall: setting allowed input port 8388 through interface eth0...
gluetun        | 2020-08-17T10:42:02.312+0200   INFO    shadowsocks configurator: starting shadowsocks server
gluetun        | 2020-08-17T10:42:02.316+0200   ERROR   shadowsocks: exec: "ss-server": executable file not found in $PATH
gluetun        | 2020-08-17T10:42:02.316+0200   INFO    shadowsocks: retrying in 1 minute
```

Configuration file:

```yml

```

Host OS: Debian Buster AMD64

**Comment by qdm12** [comment]
Well I guess it was time for #220 😉
That alpine shadowsocks dependency is really becoming a problem... Feel free to try the image `:shadowsocks`.

**Comment by Frepke** [comment]
> 
> 
> Well I guess it was time for #220 😉
> That alpine shadowsocks dependency is really becoming a problem... Feel free to try the image `:shadowsocks`. You can comment back on #211 also

Don't see the error appear in `:shadowsocks`

```
gluetun        | Running version shadowsocks built on 2020-08-16T23:26:12Z (commit fc1d960)
gluetun        | 
gluetun        | 
gluetun        | 🔧  Need help? https://github.com/qdm12/gluetun/issues/new
gluetun        | 💻  Email? quentin.mcgaw@gmail.com
gluetun        | ☕  Slack? Join from the Slack button on Github
gluetun        | 💸  Help me? https://github.com/sponsors/qdm12
gluetun        | 2020-08-17T14:09:53.706+0200   INFO    Unbound version: 1.10.1
gluetun        | 2020-08-17T14:09:53.717+0200   INFO    IPtables version: v1.8.4
gluetun        | 2020-08-17T14:09:53.777+0200   INFO    TinyProxy version: 1.10.0
gluetun        | 2020-08-17T14:09:53.866+0200   INFO    OpenVPN version: 2.4.9
gluetun        | 2020-08-17T14:09:53.870+0200   INFO    Settings summary below:
gluetun        | OpenVPN settings:
gluetun        | |--User: [redacted]
gluetun        | |--Password: [redacted]
gluetun        | |--Verbosity level: 1
gluetun        | |--Run as root: no
gluetun        | |--Surfshark settings:
gluetun        |  |--Network protocol: udp
gluetun        |  |--Region: netherlands amsterdam
gluetun        |  |--Target IP address: ������������Y.�h
gluetun        | |--Custom cipher: aes-256-gcm
gluetun        | System settings:
gluetun        | |--User ID: 1000
gluetun        | |--Group ID: 100
gluetun        | |--Timezone: europe/amsterdam
gluetun        | |--IP Status filepath: /ip
gluetun        | DNS over TLS disabled, using plaintext DNS 162.252.172.57
gluetun        | Firewall settings:
gluetun        |  |--Allowed subnets: 10.54.1.0/24
gluetun        |  |--VPN input ports: 
gluetun        | TinyProxy settings: disabled
gluetun        | ShadowSocks settings:
gluetun        |  |--Password: [redacted]
gluetun        |  |--Log: enabled
gluetun        |  |--Port: 8388
gluetun        |  |--Method: chacha20-ietf-poly1305
gluetun        | Public IP check period: 12h0m0s
gluetun        | 
gluetun        | 2020-08-17T14:09:53.872+0200   INFO    routing: default route found: interface eth0, gateway 172.17.0.1
gluetun        | 2020-08-17T14:09:53.872+0200   INFO    routing: local subnet found: 172.17.0.0/16
gluetun        | 2020-08-17T14:09:53.872+0200   INFO    openvpn configurator: checking for device /dev/net/tun
gluetun        | 2020-08-17T14:09:53.872+0200   WARN    TUN device is not available: open /dev/net/tun: no such file or directory
gluetun        | 2020-08-17T14:09:53.872+0200   INFO    openvpn configurator: creating /dev/net/tun
gluetun        | 2020-08-17T14:09:53.872+0200   INFO    firewall: enabling...
gluetun        | 2020-08-17T14:09:53.872+0200   INFO    Launching standard output merger
gluetun        | 2020-08-17T14:09:53.934+0200   INFO    firewall: enabled successfully
gluetun        | 2020-08-17T14:09:53.934+0200   INFO    firewall: setting allowed subnets through firewall...
gluetun        | 2020-08-17T14:09:53.941+0200   INFO    routing: adding 10.54.1.0/24 as route via 172.17.0.1 eth0
gluetun        | 2020-08-17T14:09:53.967+0200   INFO    dns over tls: falling back on plaintext DNS at address 162.252.172.57
gluetun        | 2020-08-17T14:09:53.967+0200   INFO    dns configurator: using DNS address 162.252.172.57 internally
gluetun        | 2020-08-17T14:09:53.967+0200   INFO    dns configurator: using DNS address 162.252.172.57 system wide
gluetun        | 2020-08-17T14:09:53.979+0200   INFO    http server: listening on 0.0.0.0:8000
gluetun        | 2020-08-17T14:09:53.981+0200   INFO    firewall: setting VPN connections through firewall...
gluetun        | 2020-08-17T14:09:53.994+0200   INFO    openvpn configurator: starting openvpn
gluetun        | 2020-08-17T14:09:53.996+0200   INFO    firewall: setting allowed input port 8388 through interface eth0...
gluetun        | 2020-08-17T14:09:54.009+0200   INFO    openvpn: OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020
gluetun        | 2020-08-17T14:09:54.009+0200   INFO    openvpn: library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
gluetun        | 2020-08-17T14:09:54.011+0200   INFO    shadowsocks: listening UDP on 0.0.0.0:8388
gluetun        | 2020-08-17T14:09:54.011+0200   INFO    shadowsocks: listening TCP on 0.0.0.0:8388
gluetun        | 2020-08-17T14:09:54.037+0200   INFO    openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]89.46.223.104:1194
gluetun        | 2020-08-17T14:09:54.037+0200   INFO    openvpn: UDP link local: (not bound)
gluetun        | 2020-08-17T14:09:54.038+0200   INFO    openvpn: UDP link remote: [AF_INET]89.46.223.104:1194
gluetun        | 2020-08-17T14:09:54.074+0200   INFO    openvpn: [nl-ams-v049.prod.surfshark.com] Peer Connection Initiated with [AF_INET]89.46.223.104:1194
gluetun        | 2020-08-17T14:10:00.376+0200   ERROR   openvpn: Unrecognized option or missing or extra parameter(s) in [PUSH-OPTIONS]:7: block-outside-dns (2.4.9)
gluetun        | 2020-08-17T14:10:00.378+0200   INFO    openvpn: TUN/TAP device tun0 opened
gluetun        | 2020-08-17T14:10:00.379+0200   INFO    openvpn: /sbin/ip link set dev tun0 up mtu 1500
gluetun        | 2020-08-17T14:10:00.384+0200   INFO    openvpn: /sbin/ip addr add dev tun0 10.8.8.3/24 broadcast 10.8.8.255
gluetun        | 2020-08-17T14:10:00.401+0200   INFO    openvpn: UID set to nonrootuser
gluetun        | 2020-08-17T14:10:00.401+0200   INFO    openvpn: Initialization Sequence Completed
gluetun        | 2020-08-17T14:10:00.402+0200   INFO    routing: default route found: interface eth0, gateway 172.17.0.1
gluetun        | 2020-08-17T14:10:00.403+0200   INFO    Gateway VPN IP address: 89.46.223.104
gluetun        | 2020-08-17T14:10:00.404+0200   INFO    dns over tls: not restarting because disabled
gluetun        | 2020-08-17T14:10:00.650+0200   INFO    ip getter: Public IP address is 89.46.223.105
```

**Comment by qdm12** [comment]
Yes because I implemented shadowsocks from Scratch in Go 😄 See [ss-server](https://github.com/qdm12/ss-server) it's a container as well if you feel like testing things 😉
Might be a little buggy though, and haven't tested UDP yet.

**Comment by qdm12** [comment]
Oops realised I messed up the Dockerfile yesterday, I removed the shadowsocks server install...
I fixed it now: [build](https://github.com/qdm12/gluetun/runs/996156976)

**Comment by Frepke** [comment]
runs like a charm again 👍
